client/modules/IDE/components/Header/Toolbar.jsx

```diff
@@ -1,253 +1,192 @@
-import * as preferenceActions from '../../actions/preferences';
-import * as projectActions from '../../actions/project';
+import {
+  hideEditProjectName,
+  showEditProjectName
+} from '../../actions/project';
+import {
+  openPreferences,
+  startAccessibleSketch,
+  startSketch,
+  stopSketch
+} from '../../actions/ide';
+import {
+  setAutorefresh,
+  setGridOutput,
+  setTextOutput
+} from '../../actions/preferences';
+import { useSketchActions } from '../../hooks';
 
 import PlayIcon from '../../../../images/play.svg';
 import StopIcon from '../../../../images/stop.svg';
 import PreferencesIcon from '../../../../images/preferences.svg';
 import EditProjectNameIcon from '../../../../images/pencil.svg';
 
-class Toolbar extends React.Component {
-  constructor(props) {
-    super(props);
-    this.handleKeyPress = this.handleKeyPress.bind(this);
-    this.handleProjectNameChange = this.handleProjectNameChange.bind(this);
-    this.handleProjectNameClick = this.handleProjectNameClick.bind(this);
-    this.handleProjectNameSave = this.handleProjectNameSave.bind(this);
+const Toolbar = (props) => {
+  const { isPlaying, infiniteLoop, preferencesIsVisible } = useSelector(
+    (state) => state.ide
+  );
+  const project = useSelector((state) => state.project);
+  const { autorefresh } = useSelector((state) => state.preferences);
+  const dispatch = useDispatch();
 
-    this.state = {
-      projectNameInputValue: props.project.name
-    };
-  }
+  const { t } = useTranslation();
+  const { changeSketchName, canEditProjectName } = useSketchActions();
+
+  const projectNameInputRef = useRef();
+  const [nameInputValue, setNameInputValue] = useState(project.name);
 
-  handleKeyPress(event) {
+  function handleKeyPress(event) {
     if (event.key === 'Enter') {
-      this.props.hideEditProjectName();
-      this.projectNameInput.blur();
+      dispatch(hideEditProjectName());
+      projectNameInputRef.current.blur();
     }
   }
 
-  handleProjectNameChange(event) {
-    this.setState({ projectNameInputValue: event.target.value });
-  }
-
-  handleProjectNameClick() {
-    if (this.canEditProjectName) {
-      this.props.showEditProjectName();
+  function handleProjectNameClick() {
+    if (canEditProjectName) {
+      dispatch(showEditProjectName());
       setTimeout(() => {
-        this.projectNameInput?.focus();
+        projectNameInputRef.current?.focus();
       }, 140);
     }
   }
 
-  handleProjectNameSave() {
-    const newProjectName = this.state.projectNameInputValue.trim();
-    if (newProjectName.length === 0) {
-      this.setState({
-        projectNameInputValue: this.props.project.name
-      });
-    } else {
-      this.props.setProjectName(newProjectName);
-      this.props.hideEditProjectName();
-      if (this.props.project.id) {
-        this.props.saveProject();
-      }
+  function handleProjectNameSave() {
+    const newName = nameInputValue;
+    if (newName.length > 0) {
```

Comment:
The handling of 0 length is a bit weird in the current version. If you click outside the input it reverts the text but keeps the input open.  It seems like yours does nothing which is probably fine.

Reply:
let me know If you want me to change this